react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({
+      email: 'Type in email'
+    });
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    // console.log('test');
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log('state:', this.state);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label>
+          Email
+          <input type='email' placeholder={this.state.email} onChange={this.handleChange}/>
```

Comment:
Be sure to pass a `value` prop to your `<input>` element so that the form is fully "controlled" by React.